### PR TITLE
feat: add keyboard controls to blog search

### LINKED ISSUE #309: Add keyboard controls to blog search
Right now the blog search is mouse-only. We should update it to include keyboard controls:

1. After typing a couple characters, pressing down should move focus to the first search result and highlight it. You should be able to move to previous and next search results using up/down keys.
2. Pressing ESC should hide the dropdown and set focus back to the textbox.

### PR DESCRIPTION
**Pull request by shanpriyan**
- Added keyboard navigation of blog search results using `ArrowUp` and `ArrowDown`
- Pressing `Enter` on selected result will open the selected page
- Pressing `Esc` key will close the search results dropdown and focuses the input
- Added `⌘ + K` key binding to focus the search field

Closes #309

### CODE REVIEW

**Review by kecrily — APPROVED** [review]
LGTM, but maybe also add a shortcut for focus to the search box?

File: src/assets/js/search.js [hunk 1]
```diff
@@ -110,16 +140,46 @@ searchInput.addEventListener('keyup', function (e) {
...
+        case "Enter":
+            resultsElement.querySelector('li.selected a').click();
+            break;
+        case "Escape":
+            e.preventDefault();
+            clearSearchResults();
+            break;
+
+        default:
+            break;
```
Comment (harish-sethuraman) [inline src/assets/js/search.js:174]:
```suggestion
```

**Comment by harish-sethuraman** [comment]
> but maybe also add a shortcut for focus to the search box? 

Yea that would be nice as well. We could do a similar implementation like other docs site to focus searchbox?  We could use `⌘ + K` keybinding?

**Comment by shanpriyan** [comment]
> LGTM, but maybe also add a shortcut for focus to the search box?

Yeah we can

**Comment by kecrily** [comment]
> We could use `⌘ + K` keybinding?

This is great, when I don't know the key bindings I always try `⌘ + K` first

**Comment by shanpriyan** [comment]
> > We could use `⌘ + K` keybinding?
> 
> This is great, when I don't know the key bindings I always try `⌘ + K` first

Shall I add `⌘ + K` key binding to focus the search field?

**Comment by harish-sethuraman** [comment]
Yep Go ahead. Use `⌘ + K` for mac and relevant key binding for windows as well. Please refer to other documentation site bindings as well like beta.reactjs.org, babeljs.io

File: src/assets/js/search.js [hunk 2]
```diff
@@ -88,6 +92,30 @@ function displaySearchResults(results) {
...
+    const { offsetHeight, offsetTop } = activeElement;
+    const { offsetHeight: parentOffsetHeight, scrollTop } = scrollParent;
+
+    const isAbove = offsetTop < scrollTop;
+    const isBelow = (offsetTop + offsetHeight) > (scrollTop + parentOffsetHeight);
+
+    if (isAbove) {
+        scrollParent.scrollTo(0, offsetTop);
+    }
+    else if (isBelow) {
```
Comment (amareshsm) [inline src/assets/js/search.js:112]: Why we have else if here?  Not just else.

Comment (shanpriyan) [reply src/assets/js/search.js:112]: It is required to check if the element is below and then we need to scroll, removing that will change the scollbar position for visible elements also

**Comment by nzakas** [comment]
This is looking really good! Just a few things:

1. When the up/down arrow is pressed, it should shift focus from the textbox to the link in the dropdown (call `focus()`). Right now, the focus remains in the textbox. The Esc key should then `focus()` back to the textbox.
2. When there are no search results, the Esc key doesn't close the dropdown. It should still close the dropdown.

**Review by amareshsm — APPROVED** [review]
LGTM

**Review by harish-sethuraman — APPROVED** [review]
LGTM

**Comment by harish-sethuraman** [comment]
When I press down arrow from textbox and come down Im not able to move to textbox again with up arrow. Can we fix this? Other that this all LGTM

**Review by harish-sethuraman — COMMENTED** [review]
When I press down arrow from textbox and come down Im not able to move to textbox again with up arrow. Can we fix this? Other that this all LGTM

**Review by nzakas — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**